command/agent_test.go

```diff
@@ -2115,7 +2138,180 @@ func TestAgent_LogFile_Config(t *testing.T) {
 
 	cmd.updateConfig(f, cfg)
 
-	assert.Equal(t, "/foo/bar/juan.log", cfg.LogFile, "actual config check")
+	assert.Equal(t, "TMPDIR/juan.log", cfg.LogFile, "actual config check")
+}
+
+func TestAgent_Config_NewLogger_Default(t *testing.T) {
+	cmd := &AgentCommand{BaseCommand: &BaseCommand{}}
+	cmd.config = agentConfig.NewConfig()
+	logger, err := cmd.newLogger()
+
+	assert.NoError(t, err)
+	assert.NotNil(t, logger)
+	assert.Equal(t, hclog.Info.String(), logger.GetLevel().String())
+}
+
+func TestAgent_Config_ReloadLogLevel(t *testing.T) {
+	cmd := &AgentCommand{BaseCommand: &BaseCommand{}}
+	var err error
+	tempDir := t.TempDir()
+
+	// Load an initial config
+	hcl := strings.ReplaceAll(BasicHclConfig, "TMPDIR", tempDir)
+	configFile := populateTempFile(t, "agent-config.hcl", hcl)
+	cmd.config, err = agentConfig.LoadConfigFile(configFile.Name())
+	if err != nil {
+		t.Fatal("Cannot load config to test update/merge", err)
+	}
+
+	// Tweak the loaded config to make sure we can put log files into a temp dir
+	// and systemd log attempts work fine, this would usually happen during Run.
+	cmd.logWriter = os.Stdout
+	cmd.logger, err = cmd.newLogger()
+	if err != nil {
+		t.Fatal("logger required for systemd log messages", err)
+	}
+
+	// Sanity check
+	assert.Equal(t, "warn", cmd.config.LogLevel)
+
+	// Load a new config
+	hcl = strings.ReplaceAll(BasicHclConfig2, "TMPDIR", tempDir)
```

Comment:
Would it be possible to assert that this change has also happened?

Reply:
Could you help me understand what you mean please? Do you mean, assert that the string `TMPDIR` no longer exists in the resultant `hcl` string?

Reply:
For example, we check that the log level has changed, but we haven't checked that the old log location (starting with `"TMPDIR"`) has changed to the new log location (starting with `tempDir`). For example, something like this:

```
assert.Equal(t, "tempdir/juan.log", cmd.config.LogLocation)
assert.Equal(t, "debug", cmd.config.LogLevel)
```

Reply:
In an ideal world, I think we'd test every aspect of logging that can get updated is updated, but getting as much as we can is definitely important in case we break something in the future.

Reply:
Hmn, I might have caused some confusion. (Sorry).

It's just the log-level that gets updated via `SIGHUP` (in `reloadLogLevel`) as the `c.logger` is already created and assigned. 

I just `TMPDIR`'d other parts of the sample config to make sure nothing breaks in other tests and as we add more things for reloading extra bits of config, it was ready to go. So for example, `TestAgent_LogFile_Config` has a line that tests the log file path is loaded (so that's covered in the first instance when config is read).

Reply:
Oh! My mistake. And no worries, it's on me too. 

However, I do have a follow-up question: If we are only uploading the log level, why is this test doing the `replaceAll`? Shouldn't we only need to update the log level?

Reply:
Mainly through laziness, in that I 'borrowed' from the `server_test` code that just did `replaceAll`.  

But that is a good question, the main reason this stuff crept in was due to us doing the systemd notifications (which require a logger, and therefore the log file needs to be put somewhere that exists).

My thinking was just that if/when we allow more bits to be reloaded via `SIGHUP` things like log file path could be tested then anyway. 

`config` is only used to hold *stuff* from file/cli etc. then parsed during start-up in `Run` (and now parts are parsed/set during a `SIGHUP` reload).  So changing it out later (`c.config =`) shouldn't (I think) make a difference. For example, the `logger` is already configured with `LogConfig` so changing the stored underlying config we've pulled in from file doesn't change the settings.

Reply:
Sounds good! Thanks, and sorry for my confusion :)